pkg/controller/v1beta1/inferenceservice/reconcilers/service/service_reconciler.go

```diff
@@ -162,27 +247,30 @@ func semanticServiceEquals(desired, existing *corev1.Service) bool {
 }
 
 // Reconcile ...
-func (r *ServiceReconciler) Reconcile() (*corev1.Service, error) {
-	// reconcile Service
-	checkResult, existingService, err := r.checkServiceExist(r.client)
-	log.Info("service reconcile", "checkResult", checkResult, "err", err)
-	if err != nil {
-		return nil, err
-	}
+func (r *ServiceReconciler) Reconcile() ([]*corev1.Service, error) {
+	for _, svc := range r.ServiceList {
+		// reconcile Service
+		checkResult, _, err := r.checkServiceExist(r.client, svc)
+		log.Info("service reconcile", "checkResult", checkResult, "err", err)
+		if err != nil {
+			return nil, err
+		}
 
-	var opErr error
-	switch checkResult {
-	case constants.CheckResultCreate:
-		opErr = r.client.Create(context.TODO(), r.Service)
-	case constants.CheckResultUpdate:
-		opErr = r.client.Update(context.TODO(), r.Service)
-	default:
-		return existingService, nil
-	}
+		var opErr error
+		switch checkResult {
+		case constants.CheckResultCreate:
+			opErr = r.client.Create(context.TODO(), svc)
+		case constants.CheckResultUpdate:
+			opErr = r.client.Update(context.TODO(), svc)
+		}
 
-	if opErr != nil {
-		return nil, opErr
+		if opErr != nil {
+			return nil, opErr
+		}
 	}
-
-	return r.Service, nil
+	// Clean up head svc when head sevices are more than 3.
+	if len(r.ServiceList) > 1 {
+		r.cleanHeadSvc()
+	}
```

Comment:
Can you explain why there could be more than 3 head svc?

Reply:
Without this logic, it will keep creating headsvc when isvc is updated. When the isvc is updated, a set of head/worker node deployment will be updated and it should have a unique head svc pointing to a new head node not interfering existing service.  As you mentioned another thread, all svc will be removed when isvc removed but creating the svc unlimited does not look good so I added this logic.